sandbox/pages/SandboxCodeBlock.vue

```diff
@@ -176,18 +203,35 @@
 </template>
 
 <script setup lang="ts">
-import { inject } from 'vue'
+import { computed, ref, inject, onBeforeMount, watch, nextTick } from 'vue'
+import type { CodeBlockEventData } from '@/types'
+import useShiki from '../composables/useShiki'
 import SandboxTitleComponent from '../components/SandboxTitleComponent.vue'
 import SandboxSectionComponent from '../components/SandboxSectionComponent.vue'
 
-const code = `{
+const { createHighlighter, highlighter } = useShiki()
+
+const highlightedToggle = ref(true)
+const origLines = [1,2,3]
+const newLines = [5,6,7]
+
+const dirtyCode = ref(false)
```

Comment:
nit: I don't know what "dirty code" is -- can we rename this, and add some comments?